docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -1486,143 +1486,111 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
-
-1. The underlying library used by your extension is using the MicroProfile Metrics API directly.
-2. The underlying library uses its own way for collecting metrics and makes them available at runtime using its own API.
-3. The underlying library does not provide metrics (or there is no library at all) and you need to insert some code in the extension's codebase that will collect the metrics.
-
-What is common for all cases is that the extension should have a build-time config property that enables metrics exposure for the extension,
-it should be named `quarkus.<extension>.metrics.enabled` and be `false` by default.
-
-==== Case 1: The library uses MP Metrics
-If the library exposes metrics by itself, we don't have to do much. However, there are a few points to consider:
-
-- It should be possible to disable all metrics for an extension, which can be a bit problematic if the library registers them directly
-rather than through the unified registration mechanism in the `quarkus-smallrye-metrics-spi` module. Therefore the library should contain
-a way to turn all metrics off, for example using a library-specific system property that can be set during build time (by emitting a `SystemPropertyBuildItem`),
-and will disable metrics if at least one of the properties `quarkus.<extensionName>.metrics.enabled` and `quarkus.smallrye-metrics.extensions.enabled` is `false`.
-- It is desirable to be able to omit the MP Metrics dependency at runtime, so if possible, the library should be written in a way that
-it will still work when the MP Metrics dependencies (or at least the implementation, `io.smallrye:smallrye-metrics`) are unavailable. This can be achieved by
-wrapping all code that does something metric-related into an `if` condition that checks whether metrics integration is enabled. If the library performs injections
-of the `MetricRegistry`, which is not avoidable by introducing an `if` condition, it will unfortunately not be possible to remove the `microprofile-metrics` dependency,
-but it is still possible to avoid the need for `smallrye-metrics` (which contains the implementation class of the metric registry) by introducing a custom "no-op"
-implementation of the `MetricRegistry` which can, for example, return `null` from all its methods. This no-op implementation should be added to the application's classes
-so that it can be injected instead of the regular `io.smallrye.metrics.MetricRegistryImpl`. An example of a no-op implementation can be found at
-https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-fault-tolerance/runtime/src/main/java/io/quarkus/smallrye/faulttolerance/runtime/NoopMetricRegistry.java[NoopMetricRegistry]
+Quarkus can be configured to gather and emit metrics using either MicroProfile Metrics or Micrometer. Extensions may want to contribute their own instrumentation, and usually do so in one of the following ways :
```

Comment:
the question is whether we will merge the micrometer extension to the core repo, if not then we shouldn't be mentioning that Quarkus can use that out of the box (not yet)

Reply:
Attempt made.